src/NUnitFramework/framework/Internal/Execution/ParallelWorkItemDispatcher.cs

```diff
@@ -221,6 +226,15 @@ private void Dispatch(WorkItem work, ParallelExecutionStrategy strategy)
         {
             log.Debug("Using {0} strategy for {1}", strategy, work.Name);
 
+            // Currently, we only track CompositeWorkItems - this could be expanded
+            var composite = work as CompositeWorkItem;
+            if (composite != null)
```

Comment:
You mentioned potentially wanting to use pattern matching. Maybe you already thought about it, but if you like you can do `if (work is CompositeWorkItem composite)`.

Reply:
I've only paid attention to the new feature as it applies to catch statements. Didn't know the above syntax was possible. I'm not sure how much I like it as far as clarity goes.

Seems like moving the declaration inside the `if` saves a line and makes `composite` visible only in the following block - both good things. But having the declaration on the right-hand side of the `is` operator makes it much less visible and potentially confusing.

I'd have to study up on more examples of this kind of usage to get comfortable with it.

Reply:
No problem. The C# team made the decision in the case of the `if` statement only,due to the prevalence of feedback they collected, that `composite` would be visible (but not definitely assigned) the same as it would be if it was declared on the previous line.

More at https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching#the-is-type-pattern-expression